sherpa/astro/data.py

```diff
@@ -83,8 +84,149 @@ def _notice_resp(chans, arf, rmf):
             arf.notice(bin_mask)
 
 
+def validate_ogip_energy_ranges(rtype, label, elo, ehi, ethresh=None):
+    """Check the lo/hi values are > 0, handling common error case.
+
+    Several checks are made, to make sure the parameters follow
+    the OGIP standard. If the checks fail a DataErr is raised.
+    When ethresh is set, the case where the low-edge of the start bin
+    is zero is treated as a special case rather than an error.
+    If a replacement is made (i.e. the low edge is set to ethresh) then
+    a warning is displayed.
+
+    Parameters
+    ----------
+    rtype : {'ARF', 'RMF'}
+        The response containing the energy columns.
+    label : str
+        The identifier to use for the error message.
+    elo, ehi : numpy arrays
+        The input ENERG_LO and ENERG_HI arrays. They are assumed
+        to be one-dimensional and have the same number of elements.
+    ethresh : None or float, optional
+        If None, then elo must be greater than 0. When set, the
+        start bin can have a low-energy edge of 0; it is replaced
+        by ethresh. If set, ethresh must be greater than 0.
+
+    Returns
+    -------
+    elo, ehi : numpy arrays
+        The validated energy limits. These can be the input arrays
+        or a copy of them. At present the ehi array is the same as
+        the input array, but this may change in the future.
+
+    Notes
+    -----
+    Only some of the constraints provided by the OGIP standard are
+    checked here, since there are issues involving numerical effects
+    (e.g. when checking that two bins do not overlap), as well as
+    uncertainty over what possible  behavior is seen in released
+    data products for missions. The current set of checks are:
+
+      - ehi > elo for each bin
+      - elo is monotonic (ascending or descending)
+      - when emin is set, the lowest value in elo is >= 0,
+        otherwise it is > 0.
+      - ethresh (if set) is less than the minimum value in ENERG_HI
+
+    A failed check raises a DataErr exception.
+
+    """
+
+    if ethresh is not None and ethresh <= 0.0:
+        raise ValueError("ethresh is None or > 0")
+
+    if (elo >= ehi).any():
+        raise DataErr('ogip-error', rtype, label,
+                      'has at least one bin with ENERG_HI < ENERG_LO')
+
+    # if elo is monotonically increasing, all elements will be True
+    #                         decreasing,                      False
+    #
+    # so the sum will be number of elements or 0
+    #
+    increasing = numpy.diff(elo, n=1) > 0.0
+    nincreasing = increasing.sum()
+    if nincreasing > 0 and nincreasing != len(increasing):
+        raise DataErr('ogip-error', rtype, label,
+                      'has a non-monotonic ENERG_LO array')
+
+    if nincreasing == 0:
```

Comment:
Added in 63c2a31 or 7cb8deb.

Reply:
It's not clear whether this condition is valid (in that I believe the OGIP standard means that the energy array should always increase). I have added a test to exercise the condition just in case.